packages/next/client/page-loader.js

```diff
@@ -5,6 +5,19 @@ import { getRouteMatcher } from './../next-server/lib/router/utils/route-matcher
 import { getRouteRegex } from './../next-server/lib/router/utils/route-regex'
 import { delBasePath } from './../next-server/lib/router/router'
 
+let trustedTypesPolicy = undefined
+if (window?.trustedTypes?.createPolicy) {
+  trustedTypesPolicy = window.trustedTypes.createPolicy('next-trusted-types', {
+    // Needs security review regarding DOM XSS
+    createHTML(dirty) {
+      return dirty.replace(/</g, '&lt;') // Escapes `<` characters to prevent the creation of new HTML elements
+    },
+    createScriptURL(dirty) {
```

Comment:
So, the attack you're guarding against with createScriptUrl is an untrusted/malicious URL, which can execute arbitrary code (e.g. by loading a script from a hostile origin that exfiltrates user data, mines bitcoins, etc)

It looks like you construct the URLs in two places, at line 263 and line 103. If you are confident that the data used to construct those URLs can't come from untrusted sources, then you could have a createScriptURL function that just returns its argument, and call policy.createScriptUrl there (with comments about your assumptions).

Then, because policy isn't exported, you and your users can reduce the amount of code that needs to be subject to careful security review to just those where the policy is used in this file.